lapis/src/test/kotlin/org/genspectrum/lapis/model/AdvancedQueryFacadeTest.kt

```diff
@@ -0,0 +1,774 @@
+    }
+
+    @Test
+    fun `given a valid advancedQuery with a 'AA insertion' with stop codon and wildcard then returns SILO query`() {
+        val advancedQuery = "ins_S:501:E?*E"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(AminoAcidInsertionContains(501, "E.*\\*E", "S")))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with string metadata expression then returns SILO query`() {
+        val advancedQuery = "some_metadata=AB"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(StringEquals("some_metadata", "AB")))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with mutation and metadata expression then returns SILO query`() {
+        val advancedQuery = "(NOT some_metadata=AB) & 300G"
+
+        val result = underTest.map(advancedQuery)
+
+        val expectedResult = And(
+            NucleotideSymbolEquals(null, 300, "G"),
+            Not(StringEquals("some_metadata", "AB")),
+        )
+
+        assertThat(result, equalTo(expectedResult))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with mutation and regex metadata expression then returns SILO query`() {
+        val advancedQuery = "(some_metadata=BANGALOR AND 300G)&(some_metadata.regex='BANGALOR' OR NOT S:501Y)"
+
+        val result = underTest.map(advancedQuery)
+
+        val expectedResult = And(
+            Or(
+                Not(AminoAcidSymbolEquals("S", 501, "Y")),
+                StringSearch("some_metadata", "BANGALOR"),
+            ),
+            NucleotideSymbolEquals(null, 300, "G"),
+            StringEquals("some_metadata", "BANGALOR"),
+        )
+
+        assertThat(result, equalTo(expectedResult))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with string (with whitespace) metadata expression then returns SILO query`() {
+        val advancedQuery = "some_metadata='Democratic Republic of the Congo'"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(StringEquals("some_metadata", "Democratic Republic of the Congo")))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with string (with regex) metadata expression then returns SILO query`() {
+        val advancedQuery = "some_metadata.regex='Basel\\{1,2\\}'"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(StringSearch("some_metadata", "Basel\\{1,2\\}")))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with escaped char in regex then returns SILO query`() {
+        val advancedQuery = "some_metadata.regex='(Democratic.*Rep$'"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(StringSearch("some_metadata", "(Democratic.*Rep$")))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with date metadata expression then returns SILO query`() {
+        val advancedQuery = "date=2020-01-01"
```

Comment:
Let's also add a test for int, float and boolean (maybe to the parametrized test below?)

Reply:
Booleans still don't seem to have a test?